test/unit/search/fm_index/bi_fm_index_char_test.cpp

```diff
@@ -0,0 +1,43 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include "fm_index_test_template.hpp"
+#include "fm_index_collection_test_template.hpp"
+
+INSTANTIATE_TYPED_TEST_CASE_P(char, fm_index_test, bi_fm_index<std::vector<unsigned char>>);
+INSTANTIATE_TYPED_TEST_CASE_P(char_collection, fm_index_collection_test,
+                              bi_fm_index<std::vector<std::vector<unsigned char>>>);
+
+/*
+#ifndef NDEBUG
+TEST(char, assert_invalid_char)
+{
+    typedef bi_fm_index<std::vector<unsigned char>> bi_fm_index_t;
+
+    unsigned char c = 255;
+    std::vector<unsigned char> text{'a', 'u', ',', c, '0'};
+
+    EXPECT_DEATH(bi_fm_index_t bi_index{text}, "Assertion failed: \\(!std::ranges::any_of\\(text, \\[] \\(char_type"
+                                               " r\\) { return to_rank\\(r\\) =="
+                                               " static_cast<underlying_rank_t<char_type>>\\(255\\);"
+                                               " }\\)\\), function construct,.*");
+}
+TEST(char_collection, assert_invalid_char)
+{
+    typedef bi_fm_index<std::vector<std::vector<unsigned char>>> bi_fm_index_t;
+
+    unsigned char c = 255;
+    std::vector<std::vector<unsigned char>> text{{'a', 'b'}, {'a', 'u', ',', c, '0'}};
+
+    EXPECT_DEATH(bi_fm_index_t bi_index{text}, "Assertion failed: \\(!std::ranges::any_of\\(text \\|"
+                                               " ranges::view::join, \\[] \\(char_type r\\) { return to_rank\\(r\\) =="
+                                               " static_cast<underlying_rank_t<char_type>>\\(255\\) \\|\\|"
+                                               " to_rank\\(r\\) == static_cast<underlying_rank_t<char_type>>\\(254\\);"
+                                               " }\\)\\), function construct,.*");
+}
+#endif
+*/
```

Comment:
why commented out?

Reply:
Hm, I don't understand :thinking:

Reply:
I also do not really understand.

If I do an assert and use `EXPECT_DEATH`, the debug unit test works fine, but the coverage test will actually be terminated by the assertion.
Internally, google test spawns a new process that then executes the death test. Maybe lcov intercepts this and handles the assertion? (it works without the death test)

Anyway, I now use the lambda with the throw and it works just fine 🎉